### PR TITLE
stop filtering result while sharing

### PR DESCRIPTION
**Pull request by avinash-0007**
- Share only works by display name right now

- Search by username to share form is broken completely as there is a filterBy option in [vue-select](https://vue-select.org/api/props.html#filterby) whose return value defaults to return (label || '').toLocaleLowerCase().indexOf(search.toLocaleLowerCase()) > -1

   - label in our code is set to displayName so only search by displayName works
- Since we already get our results by searching(OCS), I don't see why we need the filterBy option in the first place
   - So I just make the return value true
   - In case we do need the option, we should make it check for both displayName and username attributes

### CODE REVIEW

**Comment by avinash-0007** [comment]
this PR is based on this PR for NcSelect https://github.com/nextcloud/calendar/pull/5645

File: src/components/SidebarTabs/SharingSearchDiv.vue [hunk 1]
```diff
@@ -30,6 +30,7 @@
 			:options="options"
 			:placeholder="t('forms', 'Search for user, group or circle …')"
 			:user-select="true"
+			:filter-by="filterResults"
```
Comment (susnux) [inline src/components/SidebarTabs/SharingSearchDiv.vue:33]: I am not sure if a simple `() => true` would not be better here.
As we do not add any logic but stub the function.

Would also help when reading the component to understand what is happening, I would say.

File: src/mixins/UserSearchMixin.js [hunk 2]
```diff
@@ -66,6 +66,17 @@ export default {
 		},
 	},
 	methods: {
+	/**
+	 * Function to filter results in NcSelect
```
Comment (susnux) [inline src/mixins/UserSearchMixin.js:70]: Otherwise (see above) if we keep this way, then please add a comment **why** you are just returning `true`.

**Review by susnux — COMMENTED** [review]
Thank you!
See comments and please fix the lint issues :)

File: src/mixins/UserSearchMixin.js (hunk 2 above)
Comment (avinash-0007) [reply src/mixins/UserSearchMixin.js]: @susnux yeah.your suggestion looks fine to me. Implemented.

**Review by Chartman123 — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Comment by Chartman123** [comment]
And squash them into one single commit :)

**Comment by avinash-0007** [comment]
@Chartman123 @susnux - created new PR here:https://github.com/nextcloud/forms/pull/1895

**Pull request closed without merging**